src/main/java/br/com/levelup/aluratech/config/security/SecurityConfigurations.java

```diff
@@ -0,0 +1,46 @@
+package br.com.levelup.aluratech.config.security;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+@EnableWebSecurity
+@Configuration
+public class SecurityConfigurations extends WebSecurityConfigurerAdapter {
+
+    private final AuthenticationService authenticationService;
+
+    public SecurityConfigurations(AuthenticationService authenticationService) {
+        this.authenticationService = authenticationService;
+    }
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(authenticationService).passwordEncoder(new BCryptPasswordEncoder());
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                .antMatchers("/api/categories/**").permitAll()
+                .antMatchers("/category/**").permitAll()
+                .antMatchers("/bGltcGEtby1jYWNoZS1kYS1hcGktYWU").permitAll()
+                .antMatchers("/admin/**").authenticated()
+                .anyRequest().authenticated()
```

Comment:
da maneira que está funciona, mas é de costume deixarmos tudo que necessita de autenticação no início, ou seja, o que é mais restrito vem primeiro:

```
.antMatchers("/admin/**").authenticated()
.antMatchers("/api/categories/**").permitAll()
.antMatchers("/category/**").permitAll()
.antMatchers("/bGltcGEtby1jYWNoZS1kYS1hcGktYWU").permitAll()
.anyRequest().authenticated()
```